lib/bolt/inventory/group2.rb

```diff
@@ -0,0 +1,312 @@
+# frozen_string_literal: true
+
+require 'bolt/inventory/group'
+
+module Bolt
+  class Inventory
+    class Group2 < Group
+      attr_accessor :name, :nodes, :aliases, :name_or_alias, :groups, :config, :rest, :facts, :vars, :features
+
+      # THESE are duplicates with the old groups for now.
+      # Regex used to validate group names and target aliases.
+      NAME_REGEX = /\A[a-z0-9_][a-z0-9_-]*\Z/.freeze
+
+      DATA_KEYS = %w[name config facts vars features].freeze
+      NODE_KEYS = DATA_KEYS + ['alias']
```

Comment:
No `uri`? It seems like this is only used for validating the yaml, and getting a `Found unexpected key(s) uri` seems wrong. Does postponing hostname validation rely on uri not being here?